core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1859,21 +1832,32 @@ public void run() {
             }
             Attributes attributes = attrBuilder.build();
 
-            boolean addressesAccepted = helper.lb.tryAcceptResolvedAddresses(
+            lastAddressesAccepted = helper.lb.tryAcceptResolvedAddresses(
                 ResolvedAddresses.newBuilder()
                     .setAddresses(servers)
                     .setAttributes(attributes)
                     .setLoadBalancingPolicyConfig(effectiveServiceConfig.getLoadBalancingConfig())
                     .build());
-
-            if (!addressesAccepted) {
-              scheduleExponentialBackOffInSyncContext();
-            }
           }
         }
       }
 
       syncContext.execute(new NamesResolved());
+
+      // If NameResolved did not assign a value to lastAddressesAccepted, we assume there was an
```

Comment:
The point of doing the callback is to workaround the fact that we can't actually determine whether the update was successful synchronously. And if `onResult()` was called from within the syncContext (which is what we're saying is a good idea long-term) then we'd be guaranteed to not know the result. (If we were okay with lying we could just return a boolean now, like we had originally planned.)

Call `resolutionResultListener` from within `NamesResolved`.